lambda_code/src/pds_service.py

```diff
@@ -22,22 +29,27 @@ def get_pds_secrets(self):
 
         return secret_object
 
-    def create_jwt(self):
+    def create_jwt(self, now: int):
         secret_object = self.get_pds_secrets()
-        current_timestamp = int(time.time())
         claims = {
             "iss": secret_object['api_key'],
             "sub": secret_object['api_key'],
             "aud": self.token_url,
-            "iat": current_timestamp,
-            "exp": current_timestamp + 30,
+            "iat": now,
+            "exp": now + self.expiry,
             "jti": str(uuid.uuid4())
         }
         return jwt.encode(claims, secret_object['private_key'], algorithm='RS512',
                           headers={"kid": secret_object['kid']})
 
     def get_access_token(self):
-        _jwt = self.create_jwt()
+        now = int(time.time())
+        cached = self.cache.get("pds_token")
+        if cached and cached["expires_at"] < now + self.expiry:
```

Comment:
Shouldn't this be `now < cached["expires_at"]`? If it adds the token expiry duration then it's comparing against the expiry time if we made a new token now which will always be more (unless the config changes somehow so expiry duration is shorter).